pkg/testing/fetcher_local.go

```diff
@@ -58,10 +58,15 @@ func (f *localFetcher) Fetch(_ context.Context, operatingSystem string, architec
 
 	mainBuildfmt := "elastic-agent-%s-%s"
 	if f.snapshotOnly && !ver.IsSnapshot() {
-		mainBuildfmt = "elastic-agent-%s-SNAPSHOT-%s"
+		if ver.Prerelease() == "" {
+			ver = semver.NewParsedSemVer(ver.Major(), ver.Minor(), ver.Patch(), ver.Prerelease()+"SNAPSHOT", ver.BuildMetadata())
```

Comment:
```suggestion
			ver = semver.NewParsedSemVer(ver.Major(), ver.Minor(), ver.Patch(), "SNAPSHOT", ver.BuildMetadata())
```

I think this would have made your explanation more obvious.

Reply:
But this is not the same thing:
The old version would transform `1.2.3-er.1` into `1.2.3-er.1-SNAPSHOT` while your suggested change would result in `1.2.3-SNAPSHOT` for the same input.
The problem comes from the fact that we handle SNAPSHOT as a separate flag while building/packaging/calculating version string so it would always result in attaching '-SNAPSHOT' to whatever version string you start from.
The code I modified keeps the same idea while transforming something that has already a non-empty prerelease into the same version but with snapshot flavor....

Reply:
On this path `ver.Prerelease() == ""` so the `ver.Prerelease()` call should return `""` and do nothing on this line.

For the second line here where `ver.Prerelease() != ""` then yes that seems correct.

Also possible I am completely misunderstanding this.

Reply:
Nope, you understood right, I switched the `if` branches in my head when I was reading the comment. 
I thought we were still talking about the comment below which gives us one more good reason to apply your suggestion :rofl:
Working too much on agent packaging and versioning does that to you  :laughing:

Reply:
Thanks, update LGTM